gateway/src/service.rs

```diff
@@ -966,13 +955,32 @@ pub struct FindProjectPayload {
 #[cfg(test)]
 pub mod tests {
     use fqdn::FQDN;
+    use shuttle_common::constants::limits::MAX_PROJECTS_DEFAULT;
 
     use super::*;
 
     use crate::task::{self, TaskResult};
     use crate::tests::{assert_err_kind, World};
     use crate::{Error, ErrorKind};
 
+    /// Verify that the account's current project count is lower than its limit.
+    async fn can_create_project(
+        service: Arc<GatewayService>,
+        account_name: &AccountName,
+        is_admin: bool,
+    ) -> bool {
+        if is_admin {
+            return true;
+        };
+
+        let project_count = service
+            .get_project_count(account_name)
+            .await
+            .expect("to get the account's project count");
+
+        project_count < MAX_PROJECTS_DEFAULT
+    }
```

Comment:
Ideally tests should not have logic that replicates the code being tested. Can we not just pass `true` and `false` into the test arguments?

Reply:
Replicating the logic means your test won't identify any errors in the replicated code.

Reply:
I didn't want to remove our only test of project limits, but I guess replicating the logic here does not resolve that. I think this should rather be tested at a higher level (handler tests), since it's now checked there. Thanks! I will remove this change.

Reply:
I added a new test for project limits. These tests could use some more work I think, but they're not run in CI, and we may be rewriting them for the new orchestrator anyway.